app/proto/communities.proto

```diff
@@ -268,3 +271,13 @@ message JoinCommunityReq {
 message LeaveCommunityReq {
   int64 community_id = 1;
 }
+
+message SearchCommunitiesReq {
+  string query = 1;
+  uint32 page_size = 2;
+}
+
+message SearchCommunitiesRes {
+  repeated Community communities = 1;
+  string next_page_token = 2;
```

Comment:
You are not supporting pagination here (which is fine), so you should remove this.